src/createServer.js

```diff
@@ -1,8 +1,161 @@
+
+      fs.readFile(filePath, (err, data) => {
+        if (err) {
+          console.error('Error loading script:', err);
+          res.statusCode = 404;
+          res.end('Not Found');
+
+          return;
+        }
+
+        res.setHeader('Content-Type', 'text/javascript');
+        res.end(data);
+      });
+    } else if (req.method === 'GET' && req.url === '/compress') {
+      res.statusCode = 400;
+      res.end('Method not allowed');
+    } else if (req.method === 'POST' && req.url === '/compress') {
+      const contentType = req.headers['content-type'];
+
+      if (!contentType || !contentType.includes('multipart/form-data')) {
+        res.statusCode = 400;
+        res.end('Invalid or missing Content-Type');
+
+        return;
+      }
+
+      let fileBuffer = null;
+      let fileName = null;
+      let compressionType = null;
+
+      const bb = busboy({ headers: req.headers });
+
+      bb.on('file', (name, file, info) => {
+        const chunks = [];
+
+        fileName = info.filename;
+
+        file.on('data', (data) => {
+          chunks.push(data);
+        });
+
+        file.on('end', () => {
+          fileBuffer = Buffer.concat(chunks);
+        });
+      });
+
+      bb.on('field', (name, value) => {
+        if (name === 'compressionType') {
+          compressionType = value;
+        }
+      });
+
+      bb.on('finish', () => {
+        if (!fileBuffer || fileBuffer.length === 0) {
+          res.statusCode = 400;
+          res.end('No file provided');
+
+          return;
+        }
+
+        if (!compressionType || !fileName) {
+          res.statusCode = 400;
+          res.end('Missing required fields');
+
+          return;
+        }
+
+        let compressionStream;
+
+        switch (compressionType) {
+          case 'gzip':
+            compressionStream = zlib.createGzip();
+            break;
+          case 'deflate':
+            compressionStream = zlib.createDeflate();
+            break;
+          case 'br':
+            compressionStream = zlib.createBrotliCompress();
+            break;
+          default:
+            console.log(`Unsupported compression type: ${compressionType}`);
+            res.statusCode = 400;
+            res.end('Unsupported compression type');
+
+            return;
+        }
+
+        const compressionExt = {
+          gzip: '.gzip',
+          deflate: '.deflate',
+          br: '.br',
+        }[compressionType];
+
+        res.statusCode = 200;
+
+        res.setHeader(
+          'Content-Disposition',
+          `attachment; filename=${fileName}${compressionExt}`,
+        );
+        res.setHeader('Content-Type', 'application/octet-stream');
+
+        const fileStream = Readable.from(fileBuffer);
+
+        fileStream
+          .pipe(compressionStream)
+          .pipe(res)
+          .on('error', (err) => {
+            console.error('Stream error:', err);
+            res.statusCode = 500;
+            res.end('Server error');
```

Comment:
Consider providing more detailed error messages or logging to help with debugging and provide better feedback to the user.